io/io/src/TBufferFile.cxx

```diff
@@ -2250,6 +2343,11 @@ void TBufferFile::WriteFastArray(void  *start, const TClass *cl, Int_t n,
 
    char *obj = (char*)start;
    if (!n) n=1;
+   else if (n < 0)
```

Comment:
Why can't we guard against the 1GB limit in this method too?

Reply:
It is not just "hard", it is impossible to know the size in the buffer of the objects; in the simplest case the object might contains a collection of (of course) unknown size at this level.